### ISSUE TITLE
Problems with Imperfecting a Long

### ISSUE THREAD

**Issue by DanielS146** [MeasuringPolyphony/mp_editor#50]
[Detractor est-Qui secuntur Long Imperfection Error MEI Parts.zip](https://github.com/MeasuringPolyphony/mp_editor/files/5046451/Detractor.est-Qui.secuntur.Long.Imperfection.Error.MEI.Parts.zip)
[Detractor est-Qui secuntur Long Imperfection Error Console Logs.log](https://github.com/MeasuringPolyphony/mp_editor/files/5046195/Detractor.est-Qui.secuntur.Long.Imperfection.Error.Console.Logs.log)

There seems to be some problems with the imperfection of the long, which may be connected to the program's interpretation of a dot and/or the subdivision of a breve into 3 semibreves. In the attached file, the first circled area contains a concealed dot within the sharp that is not imperfecting the initial long with the adjacent 3 semibreves as expected. In the second circled area, the three semibreves are imperfecting the long, which the dot should prevent.
![Problem 10 Edit](https://user-images.githubusercontent.com/69400378/89718763-8a2d5980-d98f-11ea-829a-18146c992698.jpg)

**Comment by martha-thomae** [comment]
Hi @DanielS146. Thanks for this!

Could you export the MEI Parts file and send it to me? I will run the scoring up separately to see why it is failing to detect these longas' corresponding imperfect and perfect qualities.

If you could also send me the console logs when clicking on the "Continue to Score Editor" button that would be very helpful as well. Thanks!

**Comment by DanielS146** [comment]
Hi martha-thomae. Gladly! I attached the console logs to the original Issue, but GitHub will not accept the MEI attachments. Since I am new to GitHub, how can I send them to you?

**Comment by martha-thomae** [comment]
No problem, @DanielS146!
Just zip the MEI file first and send the zip :) 

Feel free to ask any other questions about GitHub.

**Comment by martha-thomae** [comment]
@karend27:
I think I have solved the issue of the second circle of the image above. But I have a few questions for the first circled excerpt, and I want to run this by you before implementing what I think would be the solution. The thing is that I don't know if the idea I have (that will fix the immediate problem) is generalizable enough.

- So the first circled excerpt deals with the following sequence:

  `L  Sb  Sb  Sb  .  B  B  L`

  _with `L` for longa, `Sb` for semibrevis, `B` for brevis, and `.` for dot_

  So, here I have the equivalent to three breves between those two longs. And the first (and only) dot is not only showing the grouping of the semibreves equivalent to a breve but acting as a dot of imperfection for the long as well.

- Now, let's imagine that the two breves are substituted by groups of semibreves (equivalent to a breve), something like this:

  `L  Sb  Sb  Sb  .  Sb  Sb  . Sb  Sb  L`

  So, here I still have three breves between the longs. The dots act as dots of division that separate the groups of semibreves that are equivalent to a breve. **But is the first dot here also indicating an imperfection of the long (as the one in the original sequence)?** Or is it just separating these three semibreves from the following ones, indicating the first grouping of semibreves that are equivalent to a breve?

  In other words, can I assume that a dot that is one-breve away from a longa (in perfect modus) is always indicating an imperfection a.p.p.?

**Comment by martha-thomae** [comment]
> * Now, let's imagine that the two breves are substituted by groups of semibreves (equivalent to a breve), something like this:
>   `L  Sb  Sb  Sb  .  Sb  Sb  . Sb  Sb  L`

Ok, I just realized that in this case an imperfection of the first longa is not an option because you cannot alter the last breve (because it has been substituted by smaller note values). So, let's consider the following sequence instead: 

`L  Sb  Sb  Sb  .  Sb  Sb  B  L`

In addition to indicate the separation of semibreve groupings, is that dot also indicating an imperfection?

At the end, the same question:

>   So, here I still have three breves between the longs. The dots act as dots of division that separate the groups of semibreves that are equivalent to a breve. **But is the first dot here also indicating an imperfection of the long (as the one in the original sequence)?** Or is it just separating these three semibreves from the following ones, indicating the first grouping of semibreves that are equivalent to a breve?
>   In other words, can I assume that a dot that is one-breve away from a longa (in perfect modus) is always indicating an imperfection a.p.p.?

**Comment by karend27** [comment]
I agree it potentially could be ambiguous. I think though if you strictly follow Franco, in the original instance 
L Sb Sb Sb . B B L
the dot is acting as a dot of division between the breve equivalent (that is comprised of three semibreves) and the two breves that follow, forcing the third breve to be altera. The only reason for that dot to be there is to do that.
In the imaginary example you suggest:
L Sb Sb Sb . Sb Sb B L
the dot needs to separate the semibreve groupings, so I think in that instance the first long has to be perfect. Perhaps there is support from Muris in his Libellus for this when he says in the section on dots "But if the dot is placed between two semibreves, it is attributed with division of tempus"

I don't think there is a way in this notation system to have this rhythmic pattern and to ensure that the first long is understood as imperfect.
